File: auth-service/src/main/java/com/eirsteir/coffeewithme/authservice/domain/RoleType.java
package com.eirsteir.coffeewithme.authservice.domain;

import java.util.stream.Stream;

public enum RoleType {
  ROLE_USER,
  ROLE_ADMIN,
  ROLE_WRITER,
  ROLE_READER;

  public static RoleType from(String name) {
    return Stream.of(RoleType.values())
        .filter(v -> v.name().equals(name))
        .findFirst()
        .orElseThrow(IllegalArgumentException::new);
  }
}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/domain/RoleType.kt
package com.eirsteir.coffeewithme.authservice.domain

import java.util.stream.Stream

enum class RoleType {
    ROLE_USER, ROLE_ADMIN, ROLE_WRITER, ROLE_READER;

    companion object {
        fun from(name: String): RoleType {
            return Stream.of(*values())
                .filter { v: RoleType -> v.name == name }
                .findFirst()
                .orElseThrow { IllegalArgumentException() }
        }
    }
}